ui/pages/onboarding-flow/add-network-modal/index.js

```diff
@@ -33,8 +33,8 @@ export default function AddNetworkModal() {
         </Typography>
       </Box>
       <NetworksForm
-        addNewNetwork
```

Comment:
Following up from the other comment thread about `addNewNetwork`, it does seem like the network form in the onboarding context has more in common with the "add new network" version of the form.

For example, I see that `isEditing` defaults to the same value as `addNewNetwork`, i.e. the form is initialized as being "editable" in the "add new network" context. We'd similarly want it to be editable during onboarding. I see in your video that it is editable despite this being false though, so maybe I'm missing something.

Reply:
If we restore this prop, we could also undo a few more changes in the NetworkForm component (e.g. the extra condition added to the `deletable` variable assignment, and the one added in the `onCancel` function. And I guess you could undo where you moved that tracking event if you wanted to keep this minimal, though I think you're right about it being more correct now in the `else` block.

Reply:
Oh wait, maybe you unset this to avoid `setNewNetworkAdded` being called? I think you're right to do that, `setNewNetworkAdded` doesn't seem correct to call in this case.

Maybe we can leave this prop here, but update the `if` condition where `setNewNetworkAdded` is called to ensure it isn't called when `setActiveOnSubmit` is `true`. That even makes sense semantically; if we're already on a network, I don't think it's helpful to tell the user it was added (they can plainly see that already)

Reply:
Ok I've added back the `addNewNetwork` prop and kept the `setActiveOnSubmit` but am only calling `setNewNetworkAdded` when `setActiveOnSubmit` is false so we don't prompt the switch to (the newly added network) modal if we've already switched the network: https://github.com/MetaMask/metamask-extension/pull/22832/commits/534f10f4f33fc4a137491f9cfa165990b6eb9ee2